src/react-components/media-browser.js

```diff
@@ -125,8 +125,12 @@ class MediaBrowser extends Component {
       clearTimeout(this._sendQueryTimeout);
     }
 
+    const lastQuery = query;
+
     // Don't update search on every keystroke, but buffer for some ms.
     this._sendQueryTimeout = setTimeout(() => {
+      if (query !== lastQuery) return;
```

Comment:
Pretty sure this is impossible, since the value of `query` is captured by the closure, it should always remain equal to lastQuery. Did you mean to compare `lastQuery` to `this.state.query`? Even then, the `clearTimeout` should make that impossible.

Reply:
ah yes, I was trying to resolve a fairly hard to reproduce issue where the state.value was getting clobbered, in part i thought because there was some bug in logic here. you're right that i intended to use `state` though. i'll just take this out for now until i get a better repro